feat: trajectory correction with bounds

paa27 opened the pull request:
## Description
Added feature to use bounded least squares ([scipy implementation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.lsq_linear.html)) to perform trajectory correction. If user provides corrector bounds, bounded least squares is used, instead of unbounded least squares (both for SVD and Micado).

## Checklist

Mandatory: 

- [x] I have added tests to cover my changes
- [~] All the tests are passing, including my new ones -> all tests passing on test_trajectory_correcton.py
- [x] I described my changes in this PR description

Optional:

- [ ] The code I wrote follows good style practices (see [PEP 8](https://peps.python.org/pep-0008/) and [PEP 20](https://peps.python.org/pep-0020/)).
- [x] I have updated the docs in relation to my changes, if applicable -> updated relevant docstrings
- [ ] I have tested also GPU contexts

giadarol commented:
Thank you for the useful feature and implementation! 

One small change request to be consistent with the nomenclature we use in match: can you call them `limits` intstead of `bounds`?

paa27 commented:
Done :). Changes:
- Renamed `*bounds*` to `*limits*` throughout PR. 
- Updated relevant docstrings.
- Re-ran tests on test_trajectory_correcton.py -> PASSED
